src/material-experimental/mdc-chips/chip-listbox.spec.ts

```diff
@@ -800,6 +796,15 @@ describe('MDC-based MatChipListbox', () => {
   }
 });
 
+/** Creates a keydown event with the given key and an optional target element. */
+function createKeydownEvent(keyCode: number, target?: Element): KeyboardEvent {
+  const event = createKeyboardEvent('keydown', keyCode, undefined);
+  if (target !== undefined) {
+    setEventTarget(event, target);
```

Comment:
It looks like a similar block of code is repeated in multiple places. Couldn't we pass in the target as a parameter?

Reply:
That was actually one of the changes. Previously we handled it as part of `createKeyboardEvent`, but that feels wrong. This seems uncommon and specific to our chips tests, so I figured the event target overriding should be only done within the chips. We should encourage event dispatching with real bubbling enabled. Sometimes this is not possible though (as within the chips that pass the event directly to the class).

Reply:
One idea would be to share this helper across the chips test files, or move it to `cdk/testing/private`.

Reply:
I think we were overriding the target, because either the fake events didn't behave consistently with the native ones when it comes to bubbling and the target, or we didn't trust that it would. IMO it should be in the dispatch function since the consumption is exactly the same everywhere and the extra function for setting it is even more verbose.

Reply:
> didn't behave consistently with the native ones when it comes to bubbling 

They behave consistently, and if they do not, then that would either highlight an issue in the browser, or in the browsers event dispatch logic. We are using the same pattern in all other tests and it works consistently, as expected.

>  or we didn't trust that it would

Neither that was the case here. The event target was overriding with `Object.defineProperty` just because the chips test pass the event directly to an internal method, _without_ the event actually being dispatched. That's how this API actually ended up in the codebase I assume. There seem no other cases where we need it.

> the consumption is exactly the same everywhere

Could you elaborate on that? All other event helpers we have. Like `createTouchEvent`, `createMouseEvent` and `createFakeEvent` do not have support for overriding the event target (as it is specific to the chips and doesn't seem reasonably).

It's actually way more verbose right now setting the modifiers as we'd need to set `undefined` for the event target override that is not desired for all tests except for chips. Hence I consider it specific to the chips (given the test implementation).

```js
// Notice how we repeat `undefined` multiple times here. That would be the case for most of the
// keyboard tests with modifiers... Not ideal at all
createKeyboardEvent('keydown', keyCode, undefined, undefined, {shift: true})
````

Reply:
As discussed per DM, we should just clean up/refactor the chips tests to not rely on event overriding. We've agreed on making this a follow-up, but I just went through it now in this PR.